plugins/pulp_rpm/plugins/distributors/yum/publish.py

```diff
@@ -579,17 +547,57 @@ def finalize(self):
                                            self.context.checksum)
 
 
-class PublishRpmAndDrpmStepIncremental(platform_steps.UnitPublishStep):
+def create_legacy_metadata_dict(unit):
     """
-    Publish all incremental errata
+    Pulp's legacy unit model had a "metadata" attribute, which the Incremental steps below used as
+    the basis for the data that gets written to disk. This function re-creates that data structure
+    and filters out fields whose name starts with a _.
+
+    :param unit:    unit instance
+    :type  unit:    pulp.server.db.model.ContentUnit
+
+    :return:    a dictionary with keys and values that are all of the fields on the unit model,
+                minus those in the unit key and those that start with an underscore.
+    :rtype:     dict
+    """
+    field_names = filter(lambda k: not k.startswith('_'), unit.__class__._fields.keys())
```

Comment:
I did a very similar thing in my puppet conversion to mongoengine. I'd like to ask the mongoengine community what can be done to improve this.